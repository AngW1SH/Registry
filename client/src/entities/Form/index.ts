import type { IForm } from "./types/types";
import FormCard from "./ui/FormCard";
import { staticForms } from "./static/staticForms";

export type { IForm };
export { FormCard, staticForms };
