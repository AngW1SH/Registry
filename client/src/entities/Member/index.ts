import type { IMember } from "./types/types";
import { getMembersByMemberIds } from "./model/getMembersByMemberIds";
import { staticMembers } from "./static/staticMembers";

export type { IMember };
export { getMembersByMemberIds, staticMembers };
