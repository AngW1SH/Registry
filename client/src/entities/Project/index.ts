import ProjectCard from "./ui/ProjectCard";
import ProjectCardAlt from "./ui/ProjectCardAlt";
import ProjectCardWithStatus from "./ui/ProjectCardWithStatus";
import { staticProjects } from "./static/staticProjects";
import { getProjectStage } from "./utils/getProjectStage";
import {
  getProjectSingleFromDTO,
  getProjectFromDTO,
} from "./utils/getProjectFromDTO/getProjectFromDTO";
import type {
  ProjectDTO,
  IProject,
  IProjectSingle,
  IProjectSingleDTO,
} from "./types/types";
import { ProjectStage } from "./types/types";

export {
  ProjectCard,
  ProjectCardAlt,
  ProjectCardWithStatus,
  staticProjects,
  getProjectStage,
  ProjectStage,
  getProjectFromDTO,
  getProjectSingleFromDTO,
};
export type { ProjectDTO, IProject, IProjectSingle, IProjectSingleDTO };
