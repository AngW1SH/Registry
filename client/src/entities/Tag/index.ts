import TagSlider from "./ui/TagSlider";
import TagList from "./ui/TagList";

export { TagSlider, TagList };
