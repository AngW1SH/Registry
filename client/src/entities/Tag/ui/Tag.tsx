import { ButtonTransparentAlt } from "@/shared/ui";
import { FC, ReactNode } from "react";

interface TagProps {
  children: ReactNode;
  className?: string;
  selected?: boolean;
}

const Tag: FC<TagProps> = ({ children, className = "", selected = false }) => {
  if (selected)
    return (
      <ButtonTransparentAlt
        className={"w-max bg-black px-5 py-2 text-white " + className}
      >
        {children}
      </ButtonTransparentAlt>
    );

  return (
    <ButtonTransparentAlt className={"w-max px-5 py-2 " + className}>
      {children}
    </ButtonTransparentAlt>
  );
};

export default Tag;
