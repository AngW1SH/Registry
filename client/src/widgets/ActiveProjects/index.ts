import ActiveProjects from "./ui/ActiveProjects";

export { ActiveProjects };
