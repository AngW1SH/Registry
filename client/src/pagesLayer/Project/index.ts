import ProjectPage from "./ui/ProjectPage";
import { generateProjectPageMetadata } from "./ui/ProjectPage";

export { ProjectPage, generateProjectPageMetadata };
