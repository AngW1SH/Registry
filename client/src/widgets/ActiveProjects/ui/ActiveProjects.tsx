import { IProject, ProjectCard } from "@/entities/Project";
import { ITag, TagList } from "@/entities/Tag";
import {
  GetActiveProjectsByTags,
  fetchActiveProjectsData,
} from "@/features/GetActiveProjectsByTags";
import { LinkWithIcon } from "@/shared/ui";
import { FC } from "react";

interface ActiveProjectsProps {
  data?: {
    projects: IProject[];
    tags: ITag[];
  };
}

const ActiveProjects: FC<ActiveProjectsProps> = async ({ data }) => {
  data = data ? data : await fetchActiveProjectsData();

  const { tags, projects } = data;

  return (
    <div>
      <GetActiveProjectsByTags tags={tags} projects={projects} />
      <div className="pt-7" />
      <LinkWithIcon href="/">Показать ещё</LinkWithIcon>
    </div>
  );
};

export default ActiveProjects;
