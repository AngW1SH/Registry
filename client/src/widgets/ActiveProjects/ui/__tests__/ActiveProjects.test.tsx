import { render } from "@testing-library/react";
import ActiveProjects from "../ActiveProjects";
import { staticTags } from "@/entities/Tag";
import { staticProjects } from "@/entities/Project";

describe("ActiveProjects Widget UI", () => {
  test("snapshot test", async () => {
    const Component = ActiveProjects({
      data: { tags: staticTags, projects: staticProjects },
    });

    // @ts-ignore
    const { asFragment } = render(await Component);

    expect(asFragment()).toMatchSnapshot();
  });
});
