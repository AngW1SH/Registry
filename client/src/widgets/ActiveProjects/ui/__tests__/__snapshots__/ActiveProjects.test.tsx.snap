// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveProjects Widget UI snapshot test 1`] = `
<DocumentFragment>
  <div>
    <div
      class="relative px-10"
    >
      <div
        class="flex overflow-hidden"
      >
        <ul
          class="flex"
          style="margin-left: 0px; transition: margin-left 0.3s;"
        >
          <li
            class="mr-3 list-none"
          >
            <button
              class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black active w-max bg-black px-5 py-2 text-white transition-all"
              type="button"
            >
              Все
            </button>
          </li>
          <li
            class="mr-3 list-none"
            data-tagname="Общество"
          >
            <button
              class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max px-5 py-2 transition-all"
              type="button"
            >
              Общество
            </button>
          </li>
          <li
            class="mr-3 list-none"
            data-tagname="История"
          >
            <button
              class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max px-5 py-2 transition-all"
              type="button"
            >
              История
            </button>
          </li>
          <li
            class="mr-3 list-none"
            data-tagname="ИТ"
          >
            <button
              class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max px-5 py-2 transition-all"
              type="button"
            >
              ИТ
            </button>
          </li>
          <li
            class="mr-3 list-none"
            data-tagname="Медицина"
          >
            <button
              class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max px-5 py-2 transition-all"
              type="button"
            >
              Медицина
            </button>
          </li>
          <li
            class="mr-3 list-none"
            data-tagname="Социология"
          >
            <button
              class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max px-5 py-2 transition-all"
              type="button"
            >
              Социология
            </button>
          </li>
          <li
            class="mr-3 list-none"
            data-tagname="Финансы"
          >
            <button
              class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max px-5 py-2 transition-all"
              type="button"
            >
              Финансы
            </button>
          </li>
          <li
            class="mr-3 list-none"
            data-tagname="Языки"
          >
            <button
              class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max px-5 py-2 transition-all"
              type="button"
            >
              Языки
            </button>
          </li>
          <li
            class="mr-3 list-none"
            data-tagname="Генетика"
          >
            <button
              class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max px-5 py-2 transition-all"
              type="button"
            >
              Генетика
            </button>
          </li>
          <li
            class="mr-3 list-none"
            data-tagname="Медиаресурсы"
          >
            <button
              class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max px-5 py-2 transition-all"
              type="button"
            >
              Медиаресурсы
            </button>
          </li>
        </ul>
      </div>
      <button
        class="absolute left-0 top-[calc(50%-15px)] h-[30px] w-[20px] rotate-180 cursor-pointer bg-[url('/arrow-right-black.svg')] bg-contain bg-no-repeat"
        data-testid="tag-slider-arrow-left"
      />
      <button
        class="absolute right-0 top-[calc(50%-15px)] h-[30px] w-[20px] cursor-pointer bg-[url('/arrow-right-black.svg')] bg-contain bg-no-repeat"
        data-testid="tag-slider-arrow-right"
      />
    </div>
    <div
      class="pt-12"
    />
    <ul
      class="grid grid-cols-1 gap-5 lg:grid-cols-2"
    >
      <li>
        <div
          class="shadow-center-lg rounded-lg px-8 pb-5 pt-10 h-full"
        >
          <div
            class="flex justify-between"
          >
            <div
              class="w-14 whitespace-nowrap text-xs font-bold"
            >
              <div
                class="pt-1"
              />
              <span
                class="block text-3xl"
              >
                09
              </span>
              сен 2023
            </div>
            <div
              class="pr-7"
            />
            <div>
              <h2
                class="text-xl font-semibold"
              >
                Биология растений в эпоху глобальных изменений климата
              </h2>
              <div
                class="pt-11"
              />
            </div>
          </div>
          <div
            class="flex text-sm text-[#696a6d] sm:pl-[83px] lg:pl-0 xl:pl-[83px]"
          >
            <div
              class="border-[#a1a1a1] pr-8 sm:border-l sm:pl-4 lg:border-l-0 lg:pl-0 xl:border-l xl:pl-4"
            >
              <h3>
                Срок реализации
                <br />
                проекта
              </h3>
              <div
                class="pt-2"
              />
              <p>
                01.12.2023
              </p>
            </div>
            <div
              class="border-l border-[#a1a1a1] pl-4"
            >
              <h3>
                Руководитель
              </h3>
              <div
                class="pt-2"
              />
              <p>
                Иванов П. М.
              </p>
            </div>
          </div>
          <div
            class="pt-10"
          />
          <div>
            <ul
              class="flex flex-wrap gap-1 justify-end"
            >
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] bg-white px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Генетика
                </button>
              </li>
            </ul>
          </div>
        </div>
      </li>
      <li>
        <div
          class="shadow-center-lg rounded-lg px-8 pb-5 pt-10 h-full"
        >
          <div
            class="flex justify-between"
          >
            <div
              class="w-14 whitespace-nowrap text-xs font-bold"
            >
              <div
                class="pt-1"
              />
              <span
                class="block text-3xl"
              >
                07
              </span>
              сен 2023
            </div>
            <div
              class="pr-7"
            />
            <div>
              <h2
                class="text-xl font-semibold"
              >
                Изучение социально-экономических проблем современного испанского общества
              </h2>
              <div
                class="pt-11"
              />
            </div>
          </div>
          <div
            class="flex text-sm text-[#696a6d] sm:pl-[83px] lg:pl-0 xl:pl-[83px]"
          >
            <div
              class="border-[#a1a1a1] pr-8 sm:border-l sm:pl-4 lg:border-l-0 lg:pl-0 xl:border-l xl:pl-4"
            >
              <h3>
                Срок реализации
                <br />
                проекта
              </h3>
              <div
                class="pt-2"
              />
              <p>
                12.10.2023
              </p>
            </div>
            <div
              class="border-l border-[#a1a1a1] pl-4"
            >
              <h3>
                Руководитель
              </h3>
              <div
                class="pt-2"
              />
              <p>
                Новиков В. С.
              </p>
            </div>
          </div>
          <div
            class="pt-10"
          />
          <div>
            <ul
              class="flex flex-wrap gap-1 justify-end"
            >
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] bg-white px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Общество
                </button>
              </li>
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] bg-white px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Социология
                </button>
              </li>
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] bg-white px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Финансы
                </button>
              </li>
            </ul>
          </div>
        </div>
      </li>
      <li>
        <div
          class="shadow-center-lg rounded-lg px-8 pb-5 pt-10 h-full"
        >
          <div
            class="flex justify-between"
          >
            <div
              class="w-14 whitespace-nowrap text-xs font-bold"
            >
              <div
                class="pt-1"
              />
              <span
                class="block text-3xl"
              >
                01
              </span>
              фев 2023
            </div>
            <div
              class="pr-7"
            />
            <div>
              <h2
                class="text-xl font-semibold"
              >
                ГМО: история, достижения, социальные и экологические риски
              </h2>
              <div
                class="pt-11"
              />
            </div>
          </div>
          <div
            class="flex text-sm text-[#696a6d] sm:pl-[83px] lg:pl-0 xl:pl-[83px]"
          >
            <div
              class="border-[#a1a1a1] pr-8 sm:border-l sm:pl-4 lg:border-l-0 lg:pl-0 xl:border-l xl:pl-4"
            >
              <h3>
                Срок реализации
                <br />
                проекта
              </h3>
              <div
                class="pt-2"
              />
              <p>
                01.06.2023
              </p>
            </div>
            <div
              class="border-l border-[#a1a1a1] pl-4"
            >
              <h3>
                Руководитель
              </h3>
              <div
                class="pt-2"
              />
              <p>
                Кириллов И. С.
              </p>
            </div>
          </div>
          <div
            class="pt-10"
          />
          <div>
            <ul
              class="flex flex-wrap gap-1 justify-end"
            >
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] bg-white px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Медицина
                </button>
              </li>
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] bg-white px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Генетика
                </button>
              </li>
            </ul>
          </div>
        </div>
      </li>
      <li>
        <div
          class="shadow-center-lg rounded-lg px-8 pb-5 pt-10 h-full"
        >
          <div
            class="flex justify-between"
          >
            <div
              class="w-14 whitespace-nowrap text-xs font-bold"
            >
              <div
                class="pt-1"
              />
              <span
                class="block text-3xl"
              >
                09
              </span>
              сен 2023
            </div>
            <div
              class="pr-7"
            />
            <div>
              <h2
                class="text-xl font-semibold"
              >
                Современные проблемы эпидемиологии, микробиологии и гигиены
              </h2>
              <div
                class="pt-11"
              />
            </div>
          </div>
          <div
            class="flex text-sm text-[#696a6d] sm:pl-[83px] lg:pl-0 xl:pl-[83px]"
          >
            <div
              class="border-[#a1a1a1] pr-8 sm:border-l sm:pl-4 lg:border-l-0 lg:pl-0 xl:border-l xl:pl-4"
            >
              <h3>
                Срок реализации
                <br />
                проекта
              </h3>
              <div
                class="pt-2"
              />
              <p>
                10.12.2023
              </p>
            </div>
            <div
              class="border-l border-[#a1a1a1] pl-4"
            >
              <h3>
                Руководитель
              </h3>
              <div
                class="pt-2"
              />
              <p>
                Беляев Ф. П.
              </p>
            </div>
          </div>
          <div
            class="pt-10"
          />
          <div>
            <ul
              class="flex flex-wrap gap-1 justify-end"
            >
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] bg-white px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Общество
                </button>
              </li>
              <li>
                <button
                  class="cursor-pointer rounded-3xl border-2 border-black px-3 py-2 font-bold text-black w-max border border-[#9a9a9a] bg-white px-5 py-2 text-sm font-normal text-[#303030] "
                  type="button"
                >
                  Социология
                </button>
              </li>
            </ul>
          </div>
        </div>
      </li>
    </ul>
    <div
      class="pt-7"
    />
    <a
      href="/"
    >
      <button
        class="relative "
        style="padding-right: 22px;"
      >
        Показать ещё
        <div
          class="absolute right-0 top-[calc(50%-5px)]"
          style="height: 12px; width: 12px; top: calc(50% - 6px);"
        >
          <img
            alt=""
            class="object-contain"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="/link-arrow-black.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </button>
    </a>
  </div>
</DocumentFragment>
`;
